docarray/array/mixins/find.py

```diff
@@ -144,7 +146,26 @@ def find(
                 index,
                 on=None,
             )
-        from docarray import Document, DocumentArray
+
+            if return_root:
+                from docarray.array.memory import DocumentArrayInMemory
+                from docarray.helper import get_root_docs
+
+                if isinstance(self, DocumentArrayInMemory):
```

Comment:
This should be implemented using inheritance.

All `DocumentStores` should inherit from the same Mixin that implements `get_root_docs` in a way, while `inMemory` implements its own way.

This here includes weird cyclic imports